readthedocs/analytics/proxied_api.py

```diff
@@ -0,0 +1,98 @@
+"""Analytics views that are severd from the same domain as the docs."""
+
+from django.db.models import F
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from readthedocs.analytics.models import PageView
+from readthedocs.api.v2.permissions import IsAuthorizedToViewVersion
+from readthedocs.core.unresolver import unresolve_from_request
+from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.projects.models import Feature, Project
+
+
+class BaseAnalyticsView(APIView):
+
+    """
+    Track page views.
+
+    Query parameters:
+
+    - project
+    - version
+    - absolute_uri: Full path with domain.
+    """
+
+    http_method_names = ['get']
+    permission_classes = [IsAuthorizedToViewVersion]
+
+    def _get_project(self):
+        cache_key = '__cached_project'
+        project = getattr(self, cache_key, None)
+
+        if not project:
+            project_slug = self.request.GET.get('project')
+            project = get_object_or_404(Project, slug=project_slug)
+            setattr(self, cache_key, project)
+
+        return project
+
+    def _get_version(self):
+        cache_key = '__cached_version'
+        version = getattr(self, cache_key, None)
```

Comment:
Is this required? I only see one call to `_get_version`, so I understand that it will always hit the database to get this object.

Reply:
The second call happens when checking for permissions `IsAuthorizedToViewVersion`